youtube_dlc/YoutubeDL.py

```diff
@@ -2086,7 +2091,10 @@ def compatible_formats(formats):
                 except (PostProcessingError) as err:
                     self.report_error('postprocessing: %s' % str(err))
                     return
-                self.record_download_archive(info_dict)
+                must_record_download_archive = True
+
+        if must_record_download_archive or self.params.get('force_write_download_archive', False):
```

Comment:
yes, you are right. It was a mistake I made when separating this from write-link. I didn't notice the issue, because it still works correctly with `-s`. I'll fix it right away

Reply:
I have made a commit fixing this